/request: added `fetchManager`

saabi opened the pull request:
Closes #473

mindrones commented on packages/tools/request/src/fetchManager/fetchManager.js:
```diff
@@ -0,0 +1,249 @@
+
+	const {
+		abort,
+		// abortAll,
```
leftover

mindrones commented on packages/tools/request/src/fetchManager/fetchManager.spec.js:
```diff
@@ -0,0 +1,943 @@
+- {asapKeys: ['a'], nextKeys: [..], fetchUpTo: 'end'}
+- {asapKeys: ['a'], fetchUpTo: 'end'}
+
+*/
```
`fetchUpTo` is from a discussion about future features so I think this can go for now

mindrones commented on packages/tools/request/src/fetchManager/fetchManager.js:
```diff
@@ -0,0 +1,249 @@
+
+	/* output streams */
+
+	const _outData = new BehaviorSubject({});
+	const _outLoadingKeys = new BehaviorSubject([]);
+	const _outEvents = new Subject();
```
Sorting:
```suggestion
	const _outData = new BehaviorSubject({});
	const _outEvents = new Subject();
	const _outLoadingKeys = new BehaviorSubject([]);
```

mindrones commented on CHANGELOG.md:
```diff
@@ -29,6 +29,10 @@
 
+## `@svizzle/request` v0.3.6 (next)
+
+- added `fetchManager` featuring three-stage-prioritized stream-based resource downloads. As note for future developments, we agreed that this implementation is too brittle and that we should use a state machine to manage the logic rather than Rx streams.
+
```
```suggestion
- added `fetchManager` featuring three-stage-prioritized stream-based resource downloads. As a note for future developments, we agreed that this implementation is too brittle and that we should use a state machine to manage the logic rather than Rx streams.

```

mindrones commented on packages/tools/request/CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+## `@svizzle/request` v0.3.6 (next)
+
+- added `fetchManager` featuring three-stage-prioritized stream-based resource downloads. As note for future developments, we agreed that this implementation is too brittle and that we should use a state machine to manage the logic rather than Rx streams.
+
```
```suggestion
- added `fetchManager` featuring three-stage-prioritized stream-based resource downloads. As a note for future developments, we agreed that this implementation is too brittle and that we should use a state machine to manage the logic rather than Rx streams.

```

mindrones commented on packages/tools/request/src/fetchManager/README.md:
```diff
@@ -0,0 +1,100 @@
+
+## Status
+
+As note for future developments, we agreed that this implementation is too
+brittle and that we should use a state machine to manage the logic rather than
+Rx streams.
```
```suggestion
As a note for future developments, we agreed that this implementation is too
brittle and that we should use a state machine to manage the logic rather than
Rx streams.
```

mindrones reviewed (approved):
GTM after tackling minor comments below.

As a note for future developments, we agreed that this implementation is too brittle and that we should use a state machine to manage the logic rather than Rx streams.

mindrones reviewed (approved):
GTM

saabi committed:
/request: added `fetchManager`

Closes #473

As a note for future developments, we agreed that this implementation is too
brittle and that we should use a state machine to manage the logic rather than
Rx streams.
